bin/spice/pkg/api/spice_api_client.go

```diff
@@ -0,0 +1,119 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+)
+
+type SpiceUser struct {
+	Email    string `json:"email,omitempty"`
+	Username string `json:"username,omitempty"`
+}
+
+type AccessTokenResponse struct {
+	AccessToken string `json:"access_token,omitempty"`
+}
+
+type SpiceApiClient struct {
+	baseUrl string
+}
+
+func NewSpiceApiClient() *SpiceApiClient {
+	return &SpiceApiClient{}
+}
+
+func (s *SpiceApiClient) Init() error {
+	s.baseUrl = "https://spice.ai"
+
+	if os.Getenv("SPICE_BASE_URL") != "" {
+		s.baseUrl = os.Getenv("SPICE_BASE_URL")
+	}
+
+	return nil
+}
+
+func (s *SpiceApiClient) GetAuthUrl(authCode string) string {
+	return fmt.Sprintf("%s/auth/device?code=%s", s.baseUrl, authCode)
+}
+
+func (s *SpiceApiClient) GetUser(accessToken string) (user SpiceUser, err error) {
+	var spiceUser SpiceUser
+
+	request, err := http.NewRequest("GET", fmt.Sprintf("%s/api/device/user", s.baseUrl), nil)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Since we log the error in the caller - we don't need to log errors here and below.

Reply:
Removed logging